### PR TITLE
[sonos] Unit tests added for OPML request parsing

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.sonos/pom.xml [hunk 1]
```diff
@@ -14,4 +14,13 @@
...
   <name>openHAB Add-ons :: Bundles :: Sonos Binding</name>
 
+  <dependencies>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-inline</artifactId>
+      <version>3.6.28</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
```
Comment (jlaur) [inline bundles/org.openhab.binding.sonos/pom.xml:24]: We should not use **mockito-inline**, please see https://github.com/openhab/openhab-addons/pull/12061#issuecomment-1016226346.

**Review by jlaur — CHANGES_REQUESTED** [review]
For now just quick comment about usage of mockito-inline.

File: bundles/org.openhab.binding.sonos/pom.xml (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.sonos/pom.xml:24]: How can I mock a static method without that (HttpUtil.executeUrl)?

Comment (lolodomo) [reply bundles/org.openhab.binding.sonos/pom.xml:24]: I will rather move the HTTP code outside the method I want to test.

Comment (lolodomo) [reply bundles/org.openhab.binding.sonos/pom.xml:24]: Changed.

Comment (jlaur) [reply bundles/org.openhab.binding.sonos/pom.xml]: Even better.

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**